Controller/GroupsController.php

```diff
@@ -39,19 +43,31 @@ public function getConfig()
 
             if (count($groupWithSuchRefNameList) > 0) {
                 $groupWithSuchRefName = $groupWithSuchRefNameList[0];
-                $validationResult->addFieldError(
-                    'refName',
-                    T::trans(
-                        'This refName is taken. Consider use \'%groupName%\' group or change current reference name.',
-                        array('%groupName%' => $groupWithSuchRefName->getName())
-                    )
-                );
+
+                if ($groupWithSuchRefName->getId() != $group->getId()) {
+                    $validationResult->addFieldError(
+                        'refName',
+                        T::trans(
+                            'This refName is taken. Consider use \'%groupName%\' group or change current reference name.',
+                            array('%groupName%' => $groupWithSuchRefName->getName())
+                        )
+                    );
+                }
             }
 
             return $validationResult;
 
         };
 
+        $mapEntity = function (array $params, $group, DataMapperInterface $defaultMapper, ContainerInterface $container) {
+            $defaultMapper->mapData($params, $group);
+
+            if ($group->getRefName() === '') {
+                $group->setRefName(null);
```

Comment:
Can you please add a short comment here why this thing is needed - like because of unique constraint bla bla ...